beacon-chain/sync/subscriber_beacon_blocks.go

```diff
@@ -60,18 +62,65 @@ func (s *Service) beaconBlockSubscriber(ctx context.Context, msg proto.Message)
 }
 
 // reconstructAndBroadcastBlobs processes and broadcasts blob sidecars for a given beacon block.
-// This function reconstructs the blob sidecars from the EL using the block's KZG commitments,
-// broadcasts the reconstructed blobs over P2P, and saves them into the blob storage.
 func (s *Service) reconstructAndBroadcastBlobs(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) {
 	if block.Version() < version.Deneb {
 		return
+	} else if block.Version() < version.Fulu {
+		s.reconstructAndBroadcastFullBlobs(ctx, block)
+	} else {
+		s.reconstructAndBroadcastBlobsInDataColumn(ctx, block)
+	}
+}
+
+// reconstructAndBroadcastBlobsInDataColumn reconstructs and broadcasts blobs in data column format for a given beacon block, it also saves data column sidecars into the blob storage.
+func (s *Service) reconstructAndBroadcastBlobsInDataColumn(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) {
+	blockRoot, err := block.Block().HashTreeRoot()
+	if err != nil {
+		log.WithError(err).Error("Failed to calculate block root")
+		return
+	}
+
+	if s.cfg.blobStorage == nil {
+		log.Warn("Blob storage is not enabled, skip saving data column, but continue to reconstruct and broadcast blobs")
+	}
+
+	// when this function is called, it's from the time when the block is received, so in almost all situations we need to get the data column from EL instead of the blob storage.
+	sidecars, err := s.cfg.executionReconstructor.ReconstructDataColumnSidecars(ctx, block, blockRoot)
+	if err != nil {
+		log.WithError(err).Debug("Cannot reconstruct data column sidecars after receiving the block")
+		return
 	}
 
-	// TODO: Apply the equivalent strategy for data columns.
-	if block.Version() >= version.Fulu {
+	nodeID := s.cfg.p2p.NodeID()
+	peerdas.CustodyGroupCountMut.RLock()
+	defer peerdas.CustodyGroupCountMut.RUnlock()
+	cgc := peerdas.ActualCustodyGroupCount()
+	info, _, err := peerdas.Info(nodeID, cgc)
+	if err != nil {
+		log.WithError(err).Error("Failed to get peer info")
 		return
 	}
```

Comment:
Feel like this could potentially be extracted into a helper function, would see it be used pretty often